packages/blocks/src/api/registration.js

```diff
@@ -98,10 +101,23 @@ export function registerBlockType( name, settings ) {
 		return;
 	}
 
+	const preFilterSettings = { ...settings };
 	settings = applyFilters( 'blocks.registerBlockType', settings, name );
 
 	if ( settings.deprecated ) {
-		settings.deprecated = settings.deprecated.map( ( deprecation ) => applyFilters( 'blocks.registerBlockType', deprecation, name ) );
+		settings.deprecated = settings.deprecated.map( ( deprecation ) =>
+			pick(
+				applyFilters(
+					'blocks.registerBlockType',
+					{
```

Comment:
Between this and the shallow clone above, there's some added overhead in the registration of each block. I don't think it will have a substantial impact, but noting since it impacts page load. We could potentially optimize this to skip the `Array#map` (and also the `preFilterSettings` assignment above) if `! hasFilter( 'blocks.registerBlockType' )`.

That said, now that I write this, I realize since the editor has some default filtering, it will always be true. Perhaps in the future if we refactor away from those filters, we could get some wins here.

Reply:
>Perhaps in the future if we refactor away from those filters, we could get some wins here.

That would be nice and would help in other places too probably.